### ISSUE TITLE
Dialogue model not accessible with NVDA 

### ISSUE THREAD

**Issue by rejinjosek** [exercism/exercism#6470]
Hi, 
I am using NVDA. I was trying to reveal the hints for one Python exercise by clicking on "Stuck? Reveal Hints
Opens in a modal"
But once the model appear, screen reader does not read anything. 
I have tried the same with JAWS 2019. It also does not work.

**Comment by SleeplessByte** [comment]
@iHiD @ErikSchierboom Only small changes need to be made to fix this.

1. The modal is `role="dialog"` which is correct, but inside it you want `role="document"` (so `<div dialog><div document>`). This way, a screenreader will be able to use browse mode for the content of the dialog.
2. You want to explicitly give the `h2` an id, and label the dialog using `aria-labelledby` 
3. Missing right now is a close button. If you do not want to make that visible use `sr-only`.

```html
<div role="dialog" aria-labelledby="dialog1Title" aria-describedby="dialog1Desc">
  <div role="document">
	<h2 id="dialog1Title">Your personal details were successfully updated</h2>
	<p id="dialog1Desc">You can change your details at any time in the user account section.</p> 
    <button type="button">Close</button>
  </div>
</div>
```

**Comment by iHiD** [comment]
@ErikSchierboom This might be a nice first PR for Aron tomorrow.

**Comment by rejinjosek** [comment]
Hey,  great to see the quick responses. I hope it will be done ASAP 😎

**Comment by dem4ron** [comment]
> Hey, great to see the quick responses. I hope it will be done ASAP 😎

Hey, could you check it now? I've checked it with VoiceOver on Mac and NVDA on Windows. It worked well, but was easier to make VoiceOver read it for you.

**Comment by SleeplessByte** [comment]
There is no voiceover on windows btw :)

**Comment by dem4ron** [comment]
Yep, I know :) I used NVDA on Windows. What I meant in my comment, is that to me VoiceOver's usage was a lot simpler.

**Comment by rejinjosek** [comment]
Hey, Thanks for the fix.  It reads the entire text in the model when I pres NVDA Key + tab.
It would be great if I could read it line by line if needed. Any thoughts/suggestions?
Tried with both JAWS and NVDA. Both read in the same way.

**Comment by dem4ron** [comment]
Thanks for the feedback. That shall be the next improvement, then!

**Comment by SleeplessByte** [comment]
@dem4ron it's because `<div id="a11y-task-hints-description" role="document">` is the wrapper. You want `a11y-task-hints-description` only on a description. This dialog does not seem to have a decription, but only a title. You can:
- Add a `sr-only` description underneath the title and move the `id` there. For example, the graphical interface may make it apparent that there are hints for each task, which you may want to put in the visually hidden description.
- or remove the description ID and described-by ID.

Either will solve what @rejinjosek is experiencing.
